compiler/rustc_middle/src/ty/error.rs

```diff
@@ -263,7 +263,18 @@ impl<'tcx> Ty<'tcx> {
             }
             ty::Slice(ty) if ty.is_simple_ty() => format!("slice `{}`", self).into(),
             ty::Slice(_) => "slice".into(),
-            ty::RawPtr(_) => "*-ptr".into(),
+            ty::RawPtr(tymut) => {
+                let tymut_string = match tymut.mutbl {
+                    hir::Mutability::Mut => tymut.to_string(),
+                    hir::Mutability::Not => format!("const {}", tymut.ty),
+                };
+                if tymut_string != "_" && tymut.ty.is_simple_text() {
+                    format!("`*{}`", tymut_string).into()
```

Comment:
I think this works

```suggestion
                if tymut.ty.is_simple_text() {
                    self.to_string()
```

I also wonder if we should change the code below for `ty::Ref` to match

Reply:
I could try it out tomorrow if you want to

Reply:
Please do, thanks!

Reply:
It would be pretty bad, the length check is really needed here:
```diff
-          |         expected `&T`, found type parameter `T`
+          |         expected reference, found type parameter `T`
```
```diff
-          |             ----------   ^ expected `&dyn Trait`, found struct `Box`
+          |             ----------   ^ expected reference, found struct `Box`

Reply:
I'm not sure if I understand how removing the length check causes `&T` to be rendered as "reference"

Reply:
oh, it's because `ty` is not considered simple_text... ugh

Reply:
Could we modify the logic to make a ref dyn Trait be considered "simple"?

Reply:
it's more complex than that, since a `Dynamic` is a `List<Binder<ExistentialPredicate>>`

I also think that having the length check is nicer than the `is_simple_ty` check, since this exists to avoid bloating the output, so a length check will always be more accurate than a type complexity check.

Reply:
Length check is fine then. As a final follow-up, is it possible to make the `&` and `*` printing logic as similar as possible?

Reply:
`raw pointer` is a little too short for the length check imo. I'll use `const raw pointer`, even though that is not actually used, but since the type will have a `const` prefix for the pointer, that seems fair